File: 2868-continuous-subarrays/continuous-subarrays.py
class Solution(object):
    def continuousSubarrays(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        hash_map = defaultdict(int)
        low = 0
        high = 0
        total = 0

        while high < len(nums):
            hash_map[nums[high]]+=1
            while(max(hash_map.keys()) - min(hash_map.keys()) > 2):
                hash_map[nums[low]] -= 1
                if hash_map[nums[low]]==0:
                    del hash_map[nums[low]]
                low += 1
            total += (high-low+1)
            high += 1
        return total
        

File: 2868-continuous-subarrays/continuous-subarrays.cpp
class Solution {
public:
    long long continuousSubarrays(vector<int>& nums) {
        map<int,int>mp;
        long long low = 0;
        long long high = 0;
        long long total = 0;

        while(high < nums.size())
        {
            mp[nums[high]]++;

            auto mini = mp.begin();
            auto maxi = mp.end();   maxi--;

            while((*maxi).first - (*mini).first > 2)
            {
                mp[nums[low]]--;
                if(mp[nums[low]] == 0){
                    mp.erase(mp.find(nums[low]));
                }
                low = low + 1;

                mini = mp.begin();
                maxi = mp.end();    maxi--;
            }

            total += (high-low+1);
            high = high + 1;
        }

        return total;
    }
};